here-naksha-lib-psql/src/main/java/com/here/naksha/lib/psql/XyzErrorMapper.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2017-2023 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+package com.here.naksha.lib.psql;
+
+import com.here.naksha.lib.core.models.XyzError;
+import java.util.HashMap;
+import java.util.Map;
+
+public class XyzErrorMapper {
+
+  private static final Map<String, XyzError> PSQL_ERROR_MAP = new HashMap<>();
+
+  static {
+    // UNINITIALIZED
+    PSQL_ERROR_MAP.put("N0000", XyzError.EXCEPTION);
+
+    // COLLECTION_EXISTS
```

Comment:
I assume you meant FEATURE_EXISTS and not COLLECTION_EXISTS?
The reason I ask is, because attempting to create a StorageCollection which already exists, ideally shouldn't result into an error. It will be generally expected to be an idempotent operation.
Whereas for FEATURE, it does makes sense (and is necessary) to return a CONFLICT error if we try to create a feature with an ID which already exists, so consumer can decide what it wants to do with that situation, whether to COMMIT or not.

Reply:
I meant COLLECTION_EXISTS.
There are two separate operation you may use:
1. EWriteOp.PUT
2. EWriteOp.CREATE

PUT is idempotent and won't return error if object is already there.
CREATE on the other hand will return error in such case.

It acts the same for feature & collection write